python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1616,96 +1844,152 @@ def can_run_with_parent(
                 # there is no limit on the size of a single run or...
                 parent_node.backfill_policy.max_partitions_per_run is None
                 # a single run can materialize all requested parent partitions
-                or parent_node.backfill_policy.max_partitions_per_run
-                > len(asset_partitions_to_request_map[parent.asset_key])
+                or parent_node.backfill_policy.max_partitions_per_run > num_requested_parents
             )
             # all targeted parents are being requested this tick
-            and len(asset_partitions_to_request_map[parent.asset_key]) == parent_target_subset.size
+            and num_requested_parents == parent_target_subset.size
         )
     ):
-        return True, ""
-    else:
         failed_reason = (
             f"partition mapping between {parent_node.key.to_user_string()} and {candidate_node.key.to_user_string()} is not simple and "
             f"{parent_node.key.to_user_string()} does not meet requirements of: targeting the same partitions as "
             f"{candidate_node.key.to_user_string()}, have all of its partitions requested in this iteration, having "
             "a backfill policy, and that backfill policy size limit is not exceeded by adding "
             f"{candidate_node.key.to_user_string()} to the run. {candidate_node.key.to_user_string()} can be materialized once {parent_node.key.to_user_string()} is materialized."
         )
-        return False, failed_reason
+        return (
+            asset_graph_view.get_empty_subset(key=candidate_asset_key),
+            [
+                (
+                    passed_entity_subset.get_internal_value(),
```

Comment:
again a naming nit, but given the complexity of the code I just want to make sure it's as understandable as possible for future readers -- I find it confusing that we're using `"passed_entity_subset"` as the name of the thing that holds the partitions that failed the filter.

I think this would be clearer if we kept the original name here ("entity_subset_to_filter")

Same comment applies to several places in this method